The go `ekiden` command is broken.

```
$ ./ekiden --datadir /tmp/foo
ts=2018-08-02T10:53:25.591109693Z level=info module=ekiden caller=root.go:79 msg="starting ekiden node"
panic: open /tmp/foo/config/write-file-atomic-eGAt5RIu1BV8AOQDxFEQSAdiONO1k64D: no such file or directory

goroutine 1 [running]:
github.com/oasislabs/ekiden/go/vendor/github.com/tendermint/tendermint/privval.(*FilePV).save(0xc4202dc2d0)
	/home/yawning/Documents/Development/go-workspace/src/github.com/oasislabs/ekiden/go/vendor/github.com/tendermint/tendermint/privval/priv_validator.go:134 +0x10a
github.com/oasislabs/ekiden/go/vendor/github.com/tendermint/tendermint/privval.(*FilePV).Save(0xc4202dc2d0)
	/home/yawning/Documents/Development/go-workspace/src/github.com/oasislabs/ekiden/go/vendor/github.com/tendermint/tendermint/privval/priv_validator.go:120 +0x6d
github.com/oasislabs/ekiden/go/vendor/github.com/tendermint/tendermint/privval.LoadOrGenFilePV(0xc4201b0030, 0x23, 0x7ffe59454e33)
	/home/yawning/Documents/Development/go-workspace/src/github.com/oasislabs/ekiden/go/vendor/github.com/tendermint/tendermint/privval/priv_validator.go:111 +0x99
github.com/oasislabs/ekiden/go/ekiden/cmd.newTendermintService(0xc420196ac0, 0x11f1500, 0xc420196ac0, 0x8)
	/home/yawning/Documents/Development/go-workspace/src/github.com/oasislabs/ekiden/go/ekiden/cmd/tendermint.go:36 +0x178
github.com/oasislabs/ekiden/go/ekiden/cmd.nodeMain(0x17c2ba0, 0xc42020e760, 0x0, 0x2)
	/home/yawning/Documents/Development/go-workspace/src/github.com/oasislabs/ekiden/go/ekiden/cmd/root.go:123 +0x76a
github.com/oasislabs/ekiden/go/vendor/github.com/spf13/cobra.(*Command).execute(0x17c2ba0, 0xc4200c0070, 0x2, 0x2, 0x17c2ba0, 0xc4200c0070)
	/home/yawning/Documents/Development/go-workspace/src/github.com/oasislabs/ekiden/go/vendor/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/oasislabs/ekiden/go/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x17c2ba0, 0x405acc, 0xc42003a1d8, 0x0)
	/home/yawning/Documents/Development/go-workspace/src/github.com/oasislabs/ekiden/go/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/oasislabs/ekiden/go/vendor/github.com/spf13/cobra.(*Command).Execute(0x17c2ba0, 0x0, 0xc4201a6a40)
	/home/yawning/Documents/Development/go-workspace/src/github.com/oasislabs/ekiden/go/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/oasislabs/ekiden/go/ekiden/cmd.Execute()
	/home/yawning/Documents/Development/go-workspace/src/github.com/oasislabs/ekiden/go/ekiden/cmd/root.go:59 +0x2d
main.main()
	/home/yawning/Documents/Development/go-workspace/src/github.com/oasislabs/ekiden/go/ekiden/main.go:9 +0x20
```

--- a/go/ekiden/cmd/common.go
+++ b/go/ekiden/cmd/common.go
@@ -5,6 +5,7 @@ import (
 	"io"
 	"os"
 	"os/signal"
+	"path"
 	"path/filepath"
 	"syscall"
 
@@ -64,11 +65,12 @@ func initConfig() {
 func initDataDir() error {
 	const permDir = 0700
 
-	fi, err := os.Lstat(dataDir)
+	// Tendermint uses dataDir/config, so we confim or make that subdirectory.
+	fi, err := os.Lstat(path.Join(dataDir, "config"))
 	if err != nil {
 		if os.IsNotExist(err) {
 			// Make the directory.
-			if err = os.MkdirAll(dataDir, permDir); err == nil {
+			if err = os.MkdirAll(path.Join(dataDir, "config"), permDir); err == nil {
 				return nil
 			}
 		}

--- a/go/ekiden/cmd/tendermint.go
+++ b/go/ekiden/cmd/tendermint.go
@@ -1,6 +1,9 @@
 package cmd
 
 import (
+	"os"
+	"time"
+
 	"github.com/oasislabs/ekiden/go/common/logging"
 	adapter "github.com/oasislabs/ekiden/go/tendermint/abci"
 
@@ -9,6 +12,7 @@ import (
 	tendermintNode "github.com/tendermint/tendermint/node"
 	tendermintPriv "github.com/tendermint/tendermint/privval"
 	tendermintProxy "github.com/tendermint/tendermint/proxy"
+	tendermintTypes "github.com/tendermint/tendermint/types"
 )
 
 // Adapter for tendermint Nodes to be managed by the ekiden service mux
@@ -32,13 +36,35 @@ func newTendermintService(mux *adapter.ApplicationServer) (*tendermintAdapter, e
 	viper.Unmarshal(&tenderConfig)
 	tenderConfig.SetRoot(dataDir)
 
+	tendermintPV := tendermintPriv.LoadOrGenFilePV(tenderConfig.PrivValidatorFile())
+	var tenderminGenesisProvider tendermintNode.GenesisDocProvider
+	genFile := tenderConfig.GenesisFile()
+	_, err := os.Lstat(genFile)
+	if err != nil && os.IsNotExist(err) {
+		rootLog.Warn("Tendermint Genesis file not present. Running as a one-node validator.")
+		genDoc := tendermintTypes.GenesisDoc{
+			ChainID:         "0xa515",
+			GenesisTime:     time.Now(),
+			ConsensusParams: tendermintTypes.DefaultConsensusParams(),
+		}
+		genDoc.Validators = []tendermintTypes.GenesisValidator{{
+			PubKey: tendermintPV.GetPubKey(),
+			Power:  10,
+		}}
+		tenderminGenesisProvider = func() (*tendermintTypes.GenesisDoc, error) {
+			return &genDoc, nil
+		}
+	} else {
+		tenderminGenesisProvider = tendermintNode.DefaultGenesisDocProviderFunc(tenderConfig)
+	}
+
 	node, err := tendermintNode.NewNode(tenderConfig,
-		tendermintPriv.LoadOrGenFilePV(tenderConfig.PrivValidatorFile()),
+		tendermintPV,
 		tendermintProxy.NewLocalClientCreator(mux.Mux()),
-		tendermintNode.DefaultGenesisDocProviderFunc(tenderConfig),
+		tenderminGenesisProvider,
 		tendermintNode.DefaultDBProvider,
 		tendermintNode.DefaultMetricsProvider,
-		&adapter.LogAdapter{Logger: logging.GetLogger("tendermint")})
+		&adapter.LogAdapter{logging.GetLogger("tendermint")})
 	if err != nil {
 		return nil, err
 	}

